Monika After Story/game/zz_backgrounds.rpy

```diff
@@ -0,0 +1,518 @@
+                self.image_overcast_night = image_overcast_night
+            else:
+                self.image_overcast_night = image_night
+
+            #Snow handling
+            if image_snow_day:
+                self.image_snow_day = image_snow_day
+            else:
+                self.image_snow_day = image_day
+
+            if image_snow_night:
+                self.image_snow_night = image_snow_night
+            else:
+                self.image_snow_night = image_night
+
+            #Then the other props
+            self.hide_calendar = hide_calendar
+            self.hide_masks = hide_masks
+
+            #Progressive handling
+            if disable_progressive is None:
+                self.disable_progressive = hide_masks
+            else:
+                self.disable_progressive = disable_progressive
+
+            self.unlocked = unlocked
+            self.entry_pp = entry_pp
+            self.exit_pp = exit_pp
+
+            # add to background map
+            self.mas_background.BACKGROUND_MAP[background_id] = self
+
+
+        def __eq__(self, other):
+            if isinstance(other, MASBackground):
+                return self.background_id == other.background_id
+            return NotImplemented
+
+
+        def __ne__(self, other):
+            result = self.__eq__(other)
+            if result is NotImplemented:
+                return result
+            return not result
+
+
+        def entry(self, old_background):
+            """
+            Run the entry programming point
+            """
+            if self.entry_pp is not None:
+                self.entry_pp(old_background)
+
+
+        def exit(self, new_background):
+            """
+            Run the exit programming point
+            """
+            if self.exit_pp is not None:
+                self.exit_pp(new_background)
+
+
+        def fromTuple(self, data_tuple):
+            """
+            Loads data from tuple
+
+            IN:
+                data_tuple - tuple of the following format:
+                    [0]: unlocked property
+            """
+            self.unlocked = data_tuple[0]
+
+
+        def toTuple(self):
+            """
+            Converts this MASWeather object into a tuple
+
+            RETURNS: tuple of the following format:
+                [0]: unlocked property
+            """
+            return (self.unlocked,)
+
+        def getDayRoom(self, weather=None):
+            """
+            Returns the day masks to use given the conditions/availablity of present assets
+            """
+            if weather is None:
+                weather = store.mas_current_weather
+
+            if weather.precip_type == mas_weather.PRECIP_TYPE_RAIN:
+                return self.image_rain_day
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_OVERCAST:
+                return self.image_overcast_day
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_SNOW:
```

Comment:
seeing that this pattern is being used (if type == constant, return value), it would be a lot nicer to use a dict mapping the constant to its appropriate image. Probably could even combine the day and night elements together like:
```python
img_map = {
    PRECIP_TYPE_RAIN: (day_image, night_image),
    ...
}
```